procedures.sql

```diff
@@ -0,0 +1,1136 @@
+AS
+BEGIN
+    UPDATE StudentStudies
+    SET registration_payment_date = @registration_payment_date
+    WHERE student_id = @student_id AND studies_id = @studies_id;
+END;
+GO
+-- EXEC PayForStudies 17, 23
+
+CREATE PROCEDURE SendGraduationCertificate
+    @student_id INT,
+    @studies_id INT,
+    @certificate_post_date DATETIME = CURRENT_TIMESTAMP
+AS
+BEGIN
+    UPDATE StudentStudies
+    SET certificate_post_date = @certificate_post_date
+    WHERE student_id = @student_id AND studies_id = @studies_id;
+END;
+GO
+-- EXEC SendGraduationCertificate 17, 23
+
+
+
+CREATE PROCEDURE AddStudentToWebinar
+    @student_id INT,
+    @webinar_id INT,
+    @payment_date DATETIME = NULL
+AS
+BEGIN
+    INSERT INTO StudentWebinar (student_id, webinar_id, payment_date)
+    VALUES (@student_id, @webinar_id, @payment_date);
+END;
+GO
+-- EXEC AddStudentToWebinar 17, 23, @payment_date = '2021-01-01 12:00:00'
+
+CREATE PROCEDURE RemoveStudentFromWebinar
+    @student_id INT,
+    @webinar_id INT
+AS
+BEGIN
+    DELETE FROM StudentWebinar
+    WHERE student_id = @student_id AND webinar_id = @webinar_id;
+END;
+GO
+-- EXEC RemoveStudentFromWebinar 17, 23
+
+CREATE PROCEDURE PayForWebinar
+    @student_id INT,
+    @webinar_id INT,
+    @payment_date DATETIME = CURRENT_TIMESTAMP
+AS
+BEGIN
+    UPDATE StudentWebinar
+    SET payment_date = @payment_date
+    WHERE student_id = @student_id AND webinar_id = @webinar_id;
+END;
+GO
+-- EXEC PayForWebinar 17, 23
+
+
+
+CREATE PROCEDURE AddStudentToSemester
+    @student_id INT,
+    @semester_id INT,
+    @payment_date DATETIME = NULL
+AS
+BEGIN
+    INSERT INTO StudentSemester (student_id, semester_id, payment_date)
+    VALUES (@student_id, @semester_id, @payment_date);
+END;
+GO
+-- EXEC AddStudentToSemester 17, 23
+
+CREATE PROCEDURE RemoveStudentFromSemester
+    @student_id INT,
+    @semester_id INT
+AS
+BEGIN
+    DELETE FROM StudentSemester
+    WHERE student_id = @student_id AND semester_id = @semester_id;
+END;
+GO
+-- EXEC RemoveStudentFromSemester 17, 23
+
+CREATE PROCEDURE PayForSemester
+    @student_id INT,
+    @semester_id INT,
+    @payment_date DATETIME = CURRENT_TIMESTAMP
+AS
+BEGIN
+    UPDATE StudentSemester
+    SET payment_date = @payment_date
+    WHERE student_id = @student_id AND semester_id = @semester_id;
+END;
+GO
+-- EXEC PayForSemester 17, 23, @payment_date = '2021-01-01 12:00:00'
+
+
+
+CREATE PROCEDURE AddStudentToInternship
+    @student_id INT,
+    @internship_id INT
+AS
+BEGIN
+    INSERT INTO InternshipStudent (student_id, internship_id)
+    VALUES (@student_id, @internship_id);
+END;
+GO
+-- EXEC AddStudentToInternship 17, 23
+
+CREATE PROCEDURE RemoveStudentFromInternship
+    @student_id INT,
+    @internship_id INT
+AS
+BEGIN
+    DELETE FROM InternshipStudent
+    WHERE student_id = @student_id AND internship_id = @internship_id;
+END;
+GO
+-- EXEC RemoveStudentFromInternship 17, 23
+
+CREATE PROCEDURE LogInternshipAttendance
```

Comment:
"log" is used to describe the action of recording or entering information, often in a chronological manner. On the other hand, "register" is used to convey the act of officially enrolling or entering information into a formal record or system.  imo be consistent

Reply:
"co dwie głowy to nie jedna"
prawda @tomaszkurcon?